addons/account_edi/models/account_move.py

```diff
@@ -135,6 +136,186 @@ def _compute_edi_show_abandon_cancel_button(self):
+                currency=invoice_line.currency_id,
+                quantity=invoice_line.quantity,
+                product=invoice_line.product_id,
+                partner=invoice_line.partner_id,
+                is_refund=invoice_line.move_id.move_type in ('in_refund', 'out_refund'),
+            )
+            tax_values_list = invoice_lines_tax_values_dict[invoice_line] = []
+            rate = abs(invoice_line.balance) / abs(invoice_line.amount_currency) if invoice_line.amount_currency else 0.0
+            for tax_res in taxes_res['taxes']:
+                tax_values_list.append({
+                    'base_line_id': invoice_line,
+                    'tax_id': self.env['account.tax'].browse(tax_res['id']),
+                    'tax_repartition_line_id': self.env['account.tax.repartition.line'].browse(tax_res['tax_repartition_line_id']),
+                    'base_amount': sign * invoice_line.company_currency_id.round(tax_res['base'] / rate if rate else 0.0),
+                    'tax_amount': sign * invoice_line.company_currency_id.round(tax_res['amount'] / rate if rate else 0.0),
+                    'base_amount_currency': sign * tax_res['base'],
+                    'tax_amount_currency': sign * tax_res['amount'],
+                })
+        grouping_key_generator = grouping_key_generator or default_grouping_key_generator
+
+        # Apply 'filter_to_apply'.
+
+        if filter_to_apply:
+            invoice_lines_tax_values_dict = {
+                invoice_line: [x for x in tax_values_list if filter_to_apply(x)]
+                for invoice_line, tax_values_list in invoice_lines_tax_values_dict.items()
+            }
+
+        # Initialize the results dict.
+
+        invoice_global_tax_details = {
+            'base_amount': 0.0,
+            'tax_amount': 0.0,
+            'base_amount_currency': 0.0,
+            'tax_amount_currency': 0.0,
+            'tax_details': defaultdict(lambda: {
+                'base_amount': 0.0,
+                'tax_amount': 0.0,
+                'base_amount_currency': 0.0,
+                'tax_amount_currency': 0.0,
+                'group_tax_details': [],
+            }),
+            'invoice_line_tax_details': defaultdict(lambda: {
+                'base_amount': 0.0,
+                'tax_amount': 0.0,
+                'base_amount_currency': 0.0,
+                'tax_amount_currency': 0.0,
+                'tax_details': defaultdict(lambda: {
+                    'base_amount': 0.0,
+                    'tax_amount': 0.0,
+                    'base_amount_currency': 0.0,
+                    'tax_amount_currency': 0.0,
+                    'group_tax_details': [],
+                }),
+            }),
+        }
+
+        # Apply 'grouping_key_generator' to 'invoice_lines_tax_values_list' and add all values to the final results.
+
+        def add_to_results(results, grouping_key, serialized_grouping_key, tax_values):
```

Comment:
Why not taking this function outside of the current one ? Declaring it here is weird.  I like the idea of making this a function, though :)

Reply:
I don't like the idea of declarating this method directly on the model because:
- this method will never be used except by the current method
- I don't want to create an open door to weird override/hack